workflows/data_pipelines/demarches_simplifiees/repository.py

```diff
@@ -0,0 +1,177 @@
+# database/repository.py
+import sqlite3
+import logging
+from contextlib import contextmanager
+from models import (
+    DemarcheResponse,
+    Dossier,
+)
+from schema import (
+    DatabaseSchema,
+)
+
+
+class DatabaseRepository:
+    def __init__(self, db_path: str):
+        self.db_path = db_path
+
+    @contextmanager
+    def get_connection(self):
+        connection = sqlite3.connect(self.db_path)
+        try:
+            yield connection
+        finally:
+            connection.close()
+
+    def initialize_database(self):
+        with self.get_connection() as conn:
+            cursor = conn.cursor()
+            for statement in DatabaseSchema.get_schema_statements():
+                cursor.execute(statement)
+            conn.commit()
+
+    def save_demarche(self, demarche: DemarcheResponse):
+        with self.get_connection() as conn:
+            cursor = conn.cursor()
+            try:
+                # Save demarche
+                cursor.execute(
+                    """
+                    INSERT OR REPLACE INTO demarches (
+                        id, number, title, state, date_creation
+                        )
+                    VALUES (?, ?, ?, ?, ?)
+                    """,
+                    (
+                        demarche.id,
+                        demarche.number,
+                        demarche.title,
+                        demarche.state,
+                        demarche.date_creation.isoformat(),
+                    ),
+                )
+
+                # Save dossiers and related data
+                for dossier in demarche.dossiers:
+                    self._save_dossier(cursor, demarche.id, dossier)
+
+                conn.commit()
+            except Exception as e:
+                conn.rollback()
+                logging.error(f"Error saving demarche: {e}")
+                raise
+
+    def _save_dossier(self, cursor: sqlite3.Cursor, demarche_id: str, dossier: Dossier):
+        # Save dossier
+        cursor.execute( # We can use executemany instead of for looping
```

Comment:
Possibilité d'utiliser `executemany` à la place ? C'est généralement plus performant qu'une boucle for.